gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -291,9 +364,47 @@ public int read(ByteBuffer byteBuffer) throws IOException {
         bufferedContentReadOffset = bytesToWrite;
       }
     }
+
+    if (footerContent != null && position >= footerStartOffset && byteBuffer.hasRemaining()) {
+      int bytesToWrite = Math.toIntExact(min(byteBuffer.remaining(), footerSize));
```

Comment:
I tend to agree with idea, however i prefer to keep the behaviour consistent with existing logic with `BufferedContent` and JSON impl to avoid in-consistencies